code/go/internal/validator/semantic/validate_version_integrity_test.go

```diff
@@ -5,103 +5,173 @@
 package semantic
 
 import (
-	"net/url"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
-
-	"github.com/elastic/package-spec/v3/code/go/pkg/specerrors"
 )
 
-func TestValidateGithubLink(t *testing.T) {
+func TestChangelogUniqueVersions(t *testing.T) {
+	var tests = []struct {
+		title       string
+		versions    []string
+		expectedErr bool
+	}{
+		{
+			"unique changelog versions",
+			[]string{
+				"1.0.0",
+				"1.0.1",
+			},
+			false,
+		},
+		{
+			"repeated changelog versions",
+			[]string{
+				"1.0.0",
+				"1.0.1",
+				"1.0.0",
+			},
+			true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.title, func(t *testing.T) {
+			err := ensureUniqueVersions(test.versions)
+			if test.expectedErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestManifestVersionHasChangelogEntry(t *testing.T) {
 	var tests = []struct {
-		link        string
-		expectedErr error
+		title           string
+		manifestVersion string
+		versions        []string
+		expectedErr     bool
 	}{
 		{
-			"https://github.com/elastic/integrations/pull/2897",
-			nil,
+			"manifest version exists",
+			"1.0.1",
+			[]string{
+				"1.0.1",
+				"1.0.0",
+			},
+			false,
 		},
 		{
-			"https://github.com/elastic/integrations/pull/abcd",
-			errGithubIssue,
+			"manifest version does not exist",
+			"1.1.0",
+			[]string{
+				"1.0.2",
+				"1.0.1",
+				"1.0.0",
+			},
+			true,
 		},
 		{
-			"https://github.com/elastic/integrations/pull/0",
-			errGithubIssue,
+			"changelog next version",
+			"1.0.1",
+			[]string{
+				"1.1.0-next",
+				"1.0.1",
+				"1.0.0",
+			},
+			false,
 		},
 		{
-			"https://github.com/elastic/integrations/pull",
-			errGithubIssue,
+			"changelog next version manifest older version",
+			"1.0.0",
+			[]string{
+				"1.1.0-next",
+				"1.0.1",
+				"1.0.0",
+			},
+			false,
```

Comment:
This test case shows the current behaviour when using next versions in changelog (but not in manifest).
Here, the manifest version `1.0.0` is not the latest stable `1.0.1`.
Should this be allowed? Or, should it be revisited this code? @jsoriano

Reply:
I think this would be a different change, and not sure if we want to be strict on that. We can keep it as is now.

Reply:
Ok, let's keep it as is :+1: